<?xml version="1.0"?>
<launch>
  <!-- Launch file for move_base to run for one of multiple turtlebot3's -->
  
  <arg name="model" default="racecar" doc="model type [burger, waffle, waffle_pi]"/>
  <!-- Namespace and tf_prefix are assumed to be the same -->
  <arg name="robot_namespace"/>
  <arg name="move_forward_only" default="false"/>

  <group ns="$(arg robot_namespace)">
  <!-- <group> -->
    <node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen">
      <!-- Default configs form the TB3 repos -->
      <param name="base_local_planner" value="AckermannPlannerROS" />
      <rosparam file="$(find single_racecar_navigation)/param/costmap_common_params_$(arg model).yaml" command="load" ns="global_costmap" />
      <rosparam file="$(find single_racecar_navigation)/param/costmap_common_params_$(arg model).yaml" command="load" ns="local_costmap" />
      <rosparam file="$(find single_racecar_navigation)/param/local_costmap_params.yaml" command="load" />
      <rosparam file="$(find single_racecar_navigation)/param/global_costmap_params.yaml" command="load" />
      <rosparam file="$(find single_racecar_navigation)/param/move_base_params.yaml" command="load" />
      <!-- <rosparam file="$(find single_racecar_navigation)/param/ackermann/ackermann_local_planner_params_$(arg model).yaml" command="load" /> -->

      <!-- Set tf_prefix for frames explicity, overwriting defaults -->
      <param name="global_costmap/scan/sensor_frame" 
       value="$(arg robot_namespace)/base_scan"/>
      <param name="global_costmap/obstacle_layer/scan/sensor_frame"
	     value="$(arg robot_namespace)/base_scan"/>
      <param name="global_costmap/global_frame" 
       value="map"/>
      <param name="global_costmap/robot_base_frame"
	     value="$(arg robot_namespace)/base_scan"/>
      <param name="local_costmap/scan/sensor_frame" 
       value="$(arg robot_namespace)/base_scan"/>
      <param name="local_costmap/obstacle_layer/scan/sensor_frame"
	     value="$(arg robot_namespace)/base_scan"/>
      <param name="local_costmap/global_frame" 
      value="map"/>
      <param name="local_costmap/robot_base_frame"
	     value="$(arg robot_namespace)/base_scan"/>
       
      <!-- <param name="base_global_planner" value="global_planner/GlobalPlanner" />
      <param name="planner_frequency" value="1.0" />
      <param name="GlobalPlanner/use_dijkstra" value="false"/>
      <param name="GlobalPlanner/use_grid_path" value="false"/>
      <param name="GlobalPlanner/old_navfn_behavior" value="true"/>
      <param name="planner_patience" value="5.0" /> -->
      
      <param name="base_global_planner" value="hybrid_astar_global_planner/HybridAstarGlobalPlanner" />

      <!-- <param name="base_global_planner" value="smooth_global_planner/SmoothGlobalPlanner" />
      <param name="SmoothGlobalPlanner/planner_frequency" value="1.0"/>
      <param name="SmoothGlobalPlanner/planner_patience" value="5.0" />
      <param name="SmoothGlobalPlanner/file_path" value="/home/gk/Documents/multi_turtlebot3_navigation/src/single_racecar_navigation/param/smoother_config_param.yaml"/> -->
      

      <param name="base_local_planner" value="AckermannPlannerROS" />
      <!-- <param name="base_local_planner" value="dwa_local_planner/DWAPlannerROS" /> -->
      <param name="controller_frequency" value="5.0" />
      <param name="controller_patience" value="15.0" />

      <!-- Centralized map server -->
      <remap from="map" to="/map"/>
      <param name="AckermannPlannerROS/min_vel_x" value="0.0" if="$(arg move_forward_only)" />
    </node>
  </group>
  
</launch>
